### ISSUE TITLE
Add Llama.cpp benchmark experiment?

### ISSUE THREAD

**Issue by zlwu92** [mlc-ai/xgrammar#189]
Hi,

I am a beginner in LLM and am new to learn structure generation with Xgrammer.
I find that you have provided the benchmark results for Llama.cpp in the blog post and paper. 
However, I do not find the benchmark experiment in the open source Xgrammer repo: `examples/benchmark/bench_grammar_compile_mask_gen.py` (I think it should be written here?)
If so, would you please add the test code snapshot for benchmarking Llama.cpp and show how to integrate it with Xgrammer?
Thanks.

Another question is when I run the `python bench_grammar_compile_mask_gen.py --backend lmformatenforcer `, I got the following error

<img width="921" alt="Image" src="https://github.com/user-attachments/assets/37b61b9e-22ca-494b-a930-0f19cb2469d5" />
with the same dataset in the file downloaded from huggingface.

<img width="657" alt="Image" src="https://github.com/user-attachments/assets/601b7e21-ab5f-48ce-917a-489ec4ddc367" />

<img width="459" alt="Image" src="https://github.com/user-attachments/assets/b0264f16-7ae0-406c-8e57-3a0a662b192c" />

What might be the problem?

**Comment by Ubospica** [comment]
Hi @zlwu92, thanks for asking questions about beginning to use XGrammar and testing about llama.cpp.

For beginners, I would suggest following [our tutorial](https://xgrammar.mlc.ai/docs/how_to/json_generation.html) that describes how to use xgrammar and huggingface transformer to guide the generation process. It's easy to learn and a very useful application scenario.

Regrading benchmark, the benchmark of llama.cpp and its internal grammar engine was done on our own fork because we needed to measure the speed of grammar initialization and mask generation.

| show how to integrate it with Xgrammer

We do have a plan to integrate XGrammar into llama.cpp because we have a C++ API with complete features. That would come later.

Other baselines have changed a bit since we did our benchmark. We will update the script accordingly to make it work.

**Comment by zlwu92** [comment]
Thank you.

Currently, does the open-sourced xgrammer include scripts for the two benchmarking experiments (1. speed of masking logits and 2. end-to-end evaluation for the LLM inference engine efficiency in serving scenarios) in the paper or not?
